Even with macOS-12 the [error](https://dev.azure.com/numpy/numpy/_build/results?buildId=25547&view=logs&j=9d7fb089-bdae-57e1-a4ec-9025f62c646c&t=a110bc30-f03d-50a4-8dba-c2547e2417d7&l=557) of `ld: library not found for -lSystem` is still there. Here is the compiler command:
```
INFO: Fortran f77 compiler: /usr/local/bin/gfortran -Wall -g -ffixed-form -fno-second-underscore -fPIC -O3 -funroll-loops
E                                  Fortran f90 compiler: /usr/local/bin/gfortran -Wall -g -fno-second-underscore -fPIC -O3 -funroll-loops
E                                  Fortran fix compiler: /usr/local/bin/gfortran -Wall -g -ffixed-form -fno-second-underscore -Wall -g -fno-second-underscore -fPIC -O3 -funroll-loops
E                                  INFO: compile options: '-I/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/src.macosx-10.15-x86_64-3.8 -I/Users/runner/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/numpy-1.24.0.dev0+638.g596fb0dcf-py3.8-macosx-10.15-x86_64.egg/numpy/core/include -I/Users/runner/hostedtoolcache/Python/3.8.13/x64/include/python3.8 -c'
E                                  extra options: '-J/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/ -I/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/'
E                                  INFO: gfortran:f90: /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/src.macosx-10.15-x86_64-3.8/_test_abstract_interface_TestAbstractInterface_ext_module-f2pywrappers2.f90
E                                  INFO: /usr/local/bin/gfortran -Wall -g -Wall -g -undefined dynamic_lookup -bundle /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/src.macosx-10.15-x86_64-3.8/_test_abstract_interface_TestAbstractInterface_ext_modulemodule.o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/src.macosx-10.15-x86_64-3.8/fortranobject.o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmpg01ush4w/foo.o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp3ofky4jb/src.macosx-10.15-x86_64-3.8/_test_abstract_interface_TestAbstractInterface_ext_module-f2pywrappers2.o -L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin13/4.9.0 -L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin13/4.9.0/../../.. -L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin13/4.9.0/../../.. -L/Users/runner/hostedtoolcache/Python/3.8.13/x64/lib -lgfortran -o ./_test_abstract_interface_TestAbstractInterface_ext_module.cpython-38-darwin.so
E                                  ld: library not found for -lSystem

```

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -104,13 +104,6 @@ stages:
 
   - job: macOS
     pool:
-      # NOTE: at time of writing, there is a danger
-      # that using an invalid vmIMage string for macOS
-      # image silently redirects to a Windows build on Azure;
-      # for now, use the only image name officially present in
-      # the docs even though i.e., numba uses another in their
-      # azure config for mac os -- Microsoft has indicated
-      # they will patch this issue
       vmImage: 'macOS-11'
     strategy:
       maxParallel: 3
@@ -208,6 +201,8 @@ stages:
     - script: python runtests.py -n --mode=full -- -rsx --junitxml=junit/test-results.xml
       displayName: 'Run Full NumPy Test Suite'
       condition: eq(variables['USE_OPENBLAS'], '1')
+      env:
+        LIBRARY_PATH: /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/lib
     - bash: python tools/openblas_support.py --check_version
       displayName: 'Verify OpenBLAS version'
       condition: eq(variables['USE_OPENBLAS'], '1')

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -198,7 +198,11 @@ stages:
     - script: python runtests.py -g --refguide-check
       displayName: 'Run Refguide Check'
       condition: eq(variables['USE_OPENBLAS'], '1')
-    - script: python runtests.py -n --mode=full -- -rsx --junitxml=junit/test-results.xml
+    - script: |
+        echo LIBRARY_PATH ${LIBRARY_PATH}
+        echo ls LIBRARY_PATH
+        ls ${LIBRARY_PATH}
+        python runtests.py -n --mode=full -- -rsx --junitxml=junit/test-results.xml
       displayName: 'Run Full NumPy Test Suite'
       condition: eq(variables['USE_OPENBLAS'], '1')
       env:

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -200,8 +200,6 @@ stages:
       condition: eq(variables['USE_OPENBLAS'], '1')
     - script: |
         echo LIBRARY_PATH ${LIBRARY_PATH}
-        echo ls LIBRARY_PATH
-        ls ${LIBRARY_PATH}
         python runtests.py -n --mode=full -- -rsx --junitxml=junit/test-results.xml
       displayName: 'Run Full NumPy Test Suite'
       condition: eq(variables['USE_OPENBLAS'], '1')

